src/webserve_kq/Server.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Server
+// Server class stores server information and set listening socket(s) belonging to this server
+//
+
+#include "../../include/Server.hpp"
+#include "../../include/macro.hpp"
+
+
+Server::Server(std::vector<int> ports, std::string name, std::string root_dir, int kq):
+_servername(name), _root_dir(root_dir)
+{
+	if (name == "" || root_dir == "")
+		throw ERR_Server("Server setting info is wrong");
```

Comment:
When throwing an error, does the program continue to work or should it stop?

Reply:
In this case, this server can't be built, so should stop. This error will be caught in the main function, and exit will be called.